lms/djangoapps/instructor_task/tasks_helper/utils.py

```diff
@@ -47,3 +57,42 @@ def tracker_emit(report_name):
     Emits a 'report.requested' event for the given report.
     """
     tracker.emit(REPORT_REQUESTED_EVENT_NAME, {"report_type": report_name, })
+
+
+class SensitiveMessageOnReports(object):
+    """
+    Adds a sensitive data message to the reports if the waffle switch
+    display_sensitive_data_msg_for_downloads is enabled.
+
+    Due to reports being generated in different ways, each one handles their own way:
+        1. Using the CSV library directly.
+        2. Using upload_csv_to_report_store function which in turn
+           uses store_rows of DjangoStorageReportStore class where is built the CSV.
+    """
+
+    def __init__(self):
+        self.should_msg_be_displayed = waffle.switch_is_active('display_sensitive_data_msg_for_downloads')
```

Comment:
This line is causing the [extra database hit counted below](https://github.com/edx/edx-platform/pull/18366/files#diff-2c82e4fbf2d66c60a6b0bcfaf06edc54R416).

To avoid this, please change the following to a property to be fetched on demand instead of performed during instance initialization, e.g.:
```python
class SensitiveMessageOnReports(object):
    def __init__(self):
        self._should_msg_be_displayed = None

    @property
    def should_msg_be_displayed(self):
        if self._should_msg_be_displayed is None:
            self._should_msg_be_displayed = waffle.switch_is_active('display_sensitive_data_msg_for_downloads')
        return self._should_msg_be_displayed
```